frontend/shared/src/client/sites.ts

```diff
@@ -0,0 +1,219 @@
+import {client} from './client'
+import {
+  AddSiteRequest,
+  ListSitesRequest,
+  ListSitesResponse,
+  RemoveSiteRequest,
+  SitesClientImpl,
+} from './.generated/daemon/v1alpha/sites'
+import {
+  SiteClientImpl,
+  CreateInviteTokenRequest,
+  BlockAccountRequest,
+  DeleteMemberRequest,
+  GetMemberRequest,
+  Member_Role,
+  ListMembersRequest,
+  GetSiteInfoRequest,
+  ListBlockedAccountsRequest,
+  RedeemInviteTokenRequest,
+  UnblockAccountRequest,
+  UpdateSiteInfoRequest,
+  PublishRequest,
+  UnpublishRequest,
+} from './.generated/site/v1alpha/site'
+import {ListPublicationsRequest} from './.generated/documents/v1alpha/documents'
+
+export async function addSite(hostname: string, inviteToken?: string) {
```

Comment:
@ericvicenti @horacioh these functions are exactly the reason why I'm against singletons. But I have a question first, why do you wrap all the client methods in separate functions at all? Why not just use the client instance directly where it needs to be used? I see that using `SomethingSomething.fromPartial()` might be a bit annoying in the client code, but to me it's *much* easier to follow, than the combination of custom react hook + wrapper function + global singleton client + code generated client implementation.

Now back to singletons, ideally IMO, these wrapper functions shouldn't exist. But if they must exist, the problem here is that there's now way to use them with a different client, unless you change the global client (which is not always what you want). So to me, the problem with these functions is that they have huge side effect, but they don't convey that in any way besides being async.